iocore/net/P_UnixNet.h

```diff
@@ -222,33 +224,53 @@ struct PollCont : public Continuation {
+  EThread *thread      = nullptr;
+  Event *trigger_event = nullptr;
   QueM(UnixNetVConnection, NetState, read, ready_link) read_ready_list;
   QueM(UnixNetVConnection, NetState, write, ready_link) write_ready_list;
   Que(UnixNetVConnection, link) open_list;
   DList(UnixNetVConnection, cop_link) cop_list;
   ASLLM(UnixNetVConnection, NetState, read, enable_link) read_enable_list;
   ASLLM(UnixNetVConnection, NetState, write, enable_link) write_enable_list;
   Que(UnixNetVConnection, keep_alive_queue_link) keep_alive_queue;
-  uint32_t keep_alive_queue_size;
+  uint32_t keep_alive_queue_size = 0;
   Que(UnixNetVConnection, active_queue_link) active_queue;
-  uint32_t active_queue_size;
-  uint32_t max_connections_per_thread_in;
-  uint32_t max_connections_active_per_thread_in;
-
-  // configuration settings for managing the active and keep-alive queues
-  uint32_t max_connections_in;
-  uint32_t max_connections_active_in;
-  uint32_t inactive_threashold_in;
-  uint32_t transaction_no_activity_timeout_in;
-  uint32_t keep_alive_no_activity_timeout_in;
-  uint32_t default_inactivity_timeout;
-
-  int startNetEvent(int event, Event *data);
+  uint32_t active_queue_size = 0;
+
+  /// configuration settings for managing the active and keep-alive queues
+  struct Config {
+    uint32_t max_connections_in                 = 0;
+    uint32_t max_connections_active_in          = 0;
+    uint32_t inactive_threshold_in              = 0;
+    uint32_t transaction_no_activity_timeout_in = 0;
+    uint32_t keep_alive_no_activity_timeout_in  = 0;
+    uint32_t default_inactivity_timeout         = 0;
+  };
+  /** Static global config, set and updated per process.
+
+      This is updated asynchronously and then events are sent to the NetHandler instances per thread
+      to copy to the per thread config at a convenient time. Because these are updated independently
+      from the command line, the update events just copy a single value from the global to the
+      local. This mechanism relies on members being identical types.
+  */
+  static Config global_config;
+  Config config; ///< Per thread copy of the @c global_config
+  // Active and keep alive queue values that depend on other configuration values.
+  // These are never updated directly, they are computed from other config values.
+  uint32_t max_connections_per_thread_in        = 0;
+  uint32_t max_connections_active_per_thread_in = 0;
+  /// Which members of @c Config the per thread values depend on.
+  /// If one of these is updated, the per thread values must also be updated.
+  static const std::bitset<sizeof(Config) / sizeof(uint32_t)> config_value_affects_per_thread_value;
+  /// Set of thread types in which nethandlers are active.
+  /// This enables signaling the correct instances when the configuration is updated.
+  /// Event type threads that use @c NetHandler must set the corresponding bit.
+  static std::bitset<sizeof(unsigned int) * CHAR_BIT> active_thread_types;
```

Comment:
Seems like this should be std::bitset<ET_MAX + > or something.

Also:  http://en.cppreference.com/w/cpp/types/numeric_limits/digits

Reply:
I looked for that but could not find any defined limit on the number of event types, other than the bit width of `unsigned int`.